### ISSUE TITLE
How should action labels and idx_for_delta be prepared for fine-tuning?

### ISSUE THREAD

**Issue by andyhandsom6** [2toinf/X-VLA#47]
Dear authors,

Thanks for your amazing work!

I would like to do some fine-tuning based on the X-VLA-Pt model, but I am not sure whether to use absolute or delta poses as labels for SFT.

I noticed that in the function `action_slice` in `datasets/utils.py`, there is an assumption that the action label is an “absolute trajectory”, and that the transformation from absolute to delta is formed via a linear operation (`action[:, idx] -= proprio[idx]`). I am wondering whether this is correct for rotation matrices, since linear operations do not have a clear mathematical meaning on Lie groups. Additionally, how should I prepare the action labels and `idx_for_delta` in order to align with the pretraining settings as closely as possible?

I am still new to manipulation, so please forgive me if I have misunderstood any part of the method.

Thanks in advance for your help!

**Comment by CathyF9600** [comment]
Hello, you are right that directly subtracting rotations has no clear physical meaning. Rotations live on a Lie group, so a linear operation like subtraction is not mathematically correct. A more rigorous way to compute a delta rotation is to use group operations, for example $dR = R_{target}R_{current} ^ {-1}$, or, if using axis angle, quaternion, or log map representations, to compute the relative rotation in the tangent space.

That said, in many learning pipelines people still apply direct subtraction on rotation parameterizations such as Euler angles or 6D rotation features. Although this is not theoretically clean, it often works in practice because the model learns a local approximation around small motions and the representation is continuous enough for supervision.

For preparing delta action labels, the key idea is that only the indices listed `in idx_for_delta` will be converted from absolute targets into values relative to the current proprio state. Concretely, suppose the action format is

`[x, y, z, rot6d(6), gripper]`

which has 10 dimensions, and your input proprio is also absolute end effector pose. If you want to get delta only for Cartesian position `[x, y, z]`, then you set

`idx_for_delta = [0, 1, 2]`


and the code will form

`action[:, i] = action[:, i] - proprio[i]`


for those indices. Rotation and gripper stay in absolute form. For another example, please check `datasets/domain_handler/lerobotv21.py`.
